### ISSUE TITLE
Difficult to diagnose transport misconfiguration

### ISSUE THREAD

**Issue by k-wall** [kroxylicious/kroxylicious#989]
**Describe the bug**

I misconfigured by client so it was using TLS rather than PLAIN.   I tried turning on `logNetwork` but nothing was logged.

I needed to resort to TRACE level logs.   That gave me a clue that something unexpected was coming down the wire.

```
2024-02-13 18:06:07 <epollEventLoopGroup-5-1> TRACE io.kroxylicious.proxy.internal.codec.KafkaRequestDecoder:36 - ChannelHandlerContext(requestDecoder, [id: 0x5005a595, L:/10.128.0.44:19092 - R:/100.64.0.3:57098]): Frame of 369295617 bytes (321 readable)
```

So, the transport was awaiting (lots and lots) of bytes.    LogNetwork never got a chance to fire.

**Expected behavior**

It would be good if the logging could somehow reveal the problem, without having to resort to TRACE.

It also highlights some opportunities for hardening.  The Kafka Broker  enforces `socket.request.max.bytes` for the maximum size of a request.  Kroxylicious ought to have an equivalent. 

I also suspect that Kroxylicious was prepared to wait for *ever* for those 369295617 bytes! It should be timing out peers that don't communicate within a reasonable timeframe or communicate too slowly.

**YAML files and logs**

Attach or copy paste your kroxylicious configuration YAML (with sensitive details
like broker address redacted).
Attach or copy and paste relevant logs.

**Additional context**

**Comment by robobario** [comment]
issue needs to be broken up. We want to see if there is a quick fix we can land in 0.5.0 and leave hardening or other larger changes until later.

**Comment by robobario** [comment]
Created #1009 for investigating next hardening steps, will close this one as resolved by #997
